src/hub.js

```diff
@@ -819,83 +820,99 @@ document.addEventListener("DOMContentLoaded", async () => {
-            const currentMeta = current.metas[0];
+              if (!isSelf && currentMeta.presence !== meta.presence && meta.profile.displayName) {
+                addToPresenceLog({
+                  type: "entered",
+                  presence: meta.presence,
+                  name: meta.profile.displayName
+                });
+              }
 
-            if (!isSelf && currentMeta.presence !== meta.presence && meta.profile.displayName) {
-              addToPresenceLog({
-                type: "entered",
-                presence: meta.presence,
-                name: meta.profile.displayName
-              });
+              if (currentMeta.profile && meta.profile && currentMeta.profile.displayName !== meta.profile.displayName) {
+                addToPresenceLog({
+                  type: "display_name_changed",
+                  oldName: currentMeta.profile.displayName,
+                  newName: meta.profile.displayName
+                });
+              }
+            } else {
+              // New presence
+              const meta = info.metas[0];
+
+              if (meta.presence && meta.profile.displayName) {
+                addToPresenceLog({
+                  type: "join",
+                  presence: meta.presence,
+                  name: meta.profile.displayName
+                });
+              }
             }
 
-            if (currentMeta.profile && meta.profile && currentMeta.profile.displayName !== meta.profile.displayName) {
-              addToPresenceLog({
-                type: "display_name_changed",
-                oldName: currentMeta.profile.displayName,
-                newName: meta.profile.displayName
-              });
-            }
-          } else {
-            // New presence
+            scene.emit("presence_updated", { sessionId, profile: meta.profile });
+          });
+
+          hubPhxPresence.onLeave((sessionId, current, info) => {
+            if (current && current.metas.length > 0) return;
+
             const meta = info.metas[0];
 
-            if (meta.presence && meta.profile.displayName) {
+            if (meta.profile.displayName) {
               addToPresenceLog({
-                type: "join",
-                presence: meta.presence,
+                type: "leave",
                 name: meta.profile.displayName
               });
             }
-          }
-        });
-
-        hubPhxPresence.onLeave((sessionId, current, info) => {
-          if (current && current.metas.length > 0) return;
-
-          const meta = info.metas[0];
-
-          if (meta.profile.displayName) {
-            addToPresenceLog({
-              type: "leave",
-              name: meta.profile.displayName
-            });
-          }
+          });
         });
-      });
+      }
+      isInitialJoin = false;
 
       const permsToken = oauthFlowPermsToken || data.perms_token;
       hubChannel.setPermissionsFromToken(permsToken);
+
+      hubChannel.sendProfileUpdate();
```

Comment:
what's the reason for this? isn't the initial presence info going to go out on join automatically?

Reply:
Oh. Hmm, I think I did this because it seems we only update presence in reticulum after entry, not after join. I'll change reticulum instead.

Reply:
Weird. I can't repro the issue. It seems we already update presence on join, so maybe I was mistaken.